experiment/compatibility-versions/kind-upgrade.sh

```diff
@@ -0,0 +1,173 @@
+    echo "                       [--no-kproxy]  [--no-control-plane]  [--no-kubelet]"
+    echo ""
+}
+
+parse_args()
+{
+    while [ "$1" != "" ]; do
+        case $1 in
+            -n | --name )			shift
+                                          	CLUSTER_NAME=$1
+                                          	;;
+            --cni-image )	           	shift
+                                          	CNI_IMAGE=$1
+                                          	;;
+            -b | --build-mode )			shift
+                                                if [ "$1" != "docker" ] && [ "$1" != "bazel" ]; then
+                                                    echo "Invalid build mode: $1"
+                                                    usage
+                                                    exit 1
+                                                fi
+                                                BUILD_MODE=$1
+                                                ;;
+            --no-kproxy )                   	UPDATE_KUBE_PROXY=false
+                                                ;;
+            --no-kubelet )                   	UPDATE_KUBELET=false
+                                                ;;
+            --no-control-plane )               	UPDATE_CONTROL_PLANE=false
+                                                ;;
+            -h | --help )                       usage
+                                                exit
+                                                ;;
+            * )                                 usage
+                                                exit 1
+        esac
+        shift
+    done
+}
+
+parse_args $*
+
+# Set default values
+CLUSTER_NAME=${CLUSTER_NAME:-kind}
+BUILD_MODE=${BUILD_MODE:-docker}
+UPDATE_KUBE_PROXY=${UPDATE_KUBE_PROXY:-true}
+UPDATE_KUBELET=${UPDATE_KUBE_PROXY:-true}
+# TODO: we can have more granularity here
+UPDATE_CONTROL_PLANE=${UPDATE_CONTROL_PLANE:-true}
+CONTROL_PLANE_COMPONENTS="kube-apiserver kube-controller-manager kube-scheduler"
+
+# Initialize variables
+# Assume go installed
+KUBE_ROOT="."
+# KUBE_ROOT="$(go env GOPATH)/src/k8s.io/kubernetes"
+source "${KUBE_ROOT}/hack/lib/version.sh"
+kube::version::get_version_vars
+DOCKER_TAG=${KUBE_GIT_VERSION/+/_}
+DOCKER_REGISTRY=${KUBE_DOCKER_REGISTRY:-k8s.gcr.io}
+export GOFLAGS="-tags=providerless"
+export KUBE_BUILD_CONFORMANCE=n
+
+# KIND nodes
+NODES=$(kind get nodes --name ${CLUSTER_NAME})
+CONTROL_PLANE_NODES=$(kind get nodes --name ${CLUSTER_NAME} | grep control)
+WORKER_NODES=$(kind get nodes --name ${CLUSTER_NAME} | grep worker)
+
+# Main
+if [[ "$BUILD_MODE" == "docker" ]]; then
+  build_docker
+  IMAGES_PATH="${KUBE_ROOT}/_output/release-images/amd64"
+  KUBELET_BINARY=$(find ${KUBE_ROOT}/_output/ -type f -name kubelet)
+else
+  build_bazel
+  IMAGES_PATH="${KUBE_ROOT}/bazel-kubernetes/bazel-out/k8-fastbuild/bin/build"
+  KUBELET_BINARY=$(find ${KUBE_ROOT}/bazel-kubernetes/ -type f -name kubelet)
+fi
```

Comment:
we no longer have bazel

Reply:
I've removed the bazel logic, flags, and references from this script now